mongo/options/changestreamoptions.go

```diff
@@ -97,6 +101,13 @@ func (cso *ChangeStreamOptions) SetStartAfter(sa interface{}) *ChangeStreamOptio
 	return cso
 }
 
+// SetCustomOptions sets the value for the CustomOptions field. Key-value pairs should correlate with
+// desired option names and values. Values must be Marshalable.
+func (cso *ChangeStreamOptions) SetCustomOptions(co map[string]interface{}) *ChangeStreamOptions {
```

Comment:
Does it make sense to use `bson.M` here instead of `map[string]interface{}`?

Those are basically the same type, but `bson.M` suggests that the API is looking for a BSON document, which is basically true. Also it's slightly nicer to declare `bson.M{...}` vs `map[string]interface{}{...}`.

Reply:
`bson.M` does seem to be a better fit because of its ease of use and implication that we're expecting BSON-marshalable values. Switched to `bson.M` instead of `map[string]interface{}`. 

There were some concerns offline about guaranteeing ordering of these custom options in the resulting command with a `map`, but we can't even guarantee when these options will be added to the command, so users should not be relying on ordering.